python/lsst/rubintv/production/guiders.py

```diff
@@ -131,6 +168,53 @@ def waitForIngest(nExpected: int, timeout: float, expRecord: DimensionRecord, bu
         sleep(cadence)
 
 
+def getConsDbValues(guiderData: GuiderData, metrics: DataFrame, stars: DataFrame | None) -> dict[str, float]:
+    """Map the metrics to the ConsDB values.
+
+    Parameters
+    ----------
+    metrics : `pandas.DataFrame`
+        DataFrame containing the metrics.
+
+    Returns
+    -------
+    consDbValues : `dict` [`str`, `float`]
+        Dictionary mapping the ConsDB value names to their values.
+    """
+    consDbValues: dict[str, float | int] = {}
+
+    totalExpTime = cast(float, guiderData.header["guider_duration"])
+    stampExpTime = 1.0 / cast(float, guiderData.header["freq"])
+    cols = cast(int, guiderData.header["roi_cols"])
+    rows = cast(int, guiderData.header["roi_rows"])
+
+    consDbValues["guider_exp_time"] = totalExpTime
+    consDbValues["guider_stamp_exp_time"] = stampExpTime
+    consDbValues["guider_roi_cols"] = cols
+    consDbValues["guider_roi_rows"] = rows
+
+    for key, value in CONSDB_KEY_MAP.items():
+        try:
+            consDbValues[key] = float(metrics[value].values[0])
+        except (KeyError, IndexError):
+            _LOG.warning(f"Key {key} not found in metrics DataFrame columns or has no values")
+
+    expTime = float(metrics["exptime"].values[0])
+    for key, value in CONSDB_KEY_MAP_EXPTIME_SCALED.items():
+        try:
+            scaledValue = float(metrics[value].values[0]) * expTime
+            consDbValues[key] = scaledValue
+        except (KeyError, IndexError):
+            _LOG.warning(f"Key {key} not found in metrics DataFrame columns or has no values")
+
+    if stars is not None and not stars.empty:
+        consDbValues["guider_t_mean"] = np.nanmedian(stars["e1_altaz"])
+        consDbValues["guider_e1_mean"] = np.nanmedian(stars["e1_altaz"])
+        consDbValues["guider_e2_mean"] = np.nanmedian(stars["e2_altaz"])
```

Comment:
You key the as `mean`, but are computing `median`...

Reply:
Oh man, this is a good point. We probably would prefer the median (hence me coding it as such) but changing that would require a schema update in ConsDB, which is quite a big deal. Not sure what the best approach is for now. I'll think about this though. Probably write the medians, and get the values moved over with a schema update in the future.